src/main/scala/org/broadinstitute/dsde/agora/server/model/AgoraApiJsonSupport.scala

```diff
@@ -69,6 +71,32 @@ object AgoraApiJsonSupport extends DefaultJsonProtocol {
     }
   }
 
+  implicit object MethodConfigurationFormat extends RootJsonFormat[MethodConfiguration] {
+    override def write(obj: MethodConfiguration): JsValue = {
+      jsonFormat10(MethodConfiguration).write(obj)
+    }
+
+    // Mirror the default values in the MethodConfiguration case class - spray-json does not know how to use them
+    // https://stackoverflow.com/questions/15740925/what-is-a-good-way-to-handle-default-values-with-spray-json
+    override def read(json: JsValue): MethodConfiguration = {
+      val jsObject = json.asJsObject
+
+      MethodConfiguration(
+        namespace = getFieldOrFail[String](jsObject, "namespace"),
+        name = getFieldOrFail[String](jsObject, "name"),
+        rootEntityType = getFieldOrFail[String](jsObject, "rootEntityType"),
+        prerequisites = getFieldOrFail[Map[String, AttributeString]](jsObject, "prerequisites"),
+        inputs = getFieldOrFail[Map[String, AttributeString]](jsObject, "inputs"),
+        outputs = getFieldOrFail[Map[String, AttributeString]](jsObject, "outputs"),
+        methodRepoMethod = getFieldOrFail[MethodRepoMethod](jsObject, "methodRepoMethod"),
+
+        methodConfigVersion = getFieldOrDefault(jsObject, "methodConfigVersion", 1),
+        deleted = getFieldOrDefault(jsObject, "deleted", false),
+        deletedDate = getFieldOrDefault[Option[DateTime]](jsObject, "deletedDate", None)
+      )
+    }
```

Comment:
honestly, I like this version less than the version in #332bf0. This is for a few reasons:

1. The previous version gave you an error message with all the missing keys; the current version will short-circuit with an exception on the first missing key. Once you fix the first missing key, you'll suddenly get an exception on the second missing key.

2. The previous boilerplate was actually shorter:
```
old: fields("name").convertTo[String]
new: getFieldOrFail[String](jsObject, "name")

old: fields.get("deletedDate") map (_.convertTo[DateTime])
new: getFieldOrDefault[Option[DateTime]](jsObject, "deletedDate", None)
```

3. I think getting the jsObject `fields` once and using it is better than calling `getFields()` repeatedly, because `getFields` is just slightly more work under the covers. This is a micro-optimization and should not at all be the deciding factor.